packages/transformer/src/IModelExporter.ts

```diff
@@ -1145,6 +1228,117 @@ export class ChangedInstanceIds {
+   * the sourceDbChanges. This method is used in addCustomChange to add the model to the updatedIds set so that the custom element changes are exported.
+   */
+  private addModelToUpdated(elementId: Id64String) {
+    const modelId = this._db.elements.getElement(elementId).model;
+    this.handleChange(this.model, "Updated", modelId);
+  }
+
+  /** TODO: Maybe relationships only? maybe not. */
+  public getCustomRelationshipDataFromId(
+    id: Id64String,
+    type: ChangedInstanceType
+  ): ChangedInstanceCustomData | undefined {
+    if (type === "relationship") {
+      return this._entityReferenceToCustomDataMap.get(
+        EntityReferences.fromEntityType(id, ConcreteEntityTypes.Relationship)
+      );
+    }
+
+    return undefined;
+  }
+
+  /**
+   * Adds the provided change to the set of relationship changes maintained by this instance of ChangedInstanceIds.
+   * If the same ECInstanceId is seen multiple times, the changedInstanceIds will be modified accordingly, i.e. if an id 'x' was updated but now we see 'x' was deleted, we will remove 'x'
+   * from the set of updatedIds and add it to the set of deletedIds for the appropriate class type.
+   * @note This method will ONLY accept changes to relationship classes. Use 'addCustomChange' instead if your change is not pertaining to a relationship class.
+   * @note In most cases, this method does not need to be called. Its only for consumers to mimic changes as if they were found in a changeset, which should only be useful in certain cases such as the changing of filter criteria for a preexisting master branch relationship.
+   * @throws if the ecClassId is NOT a relationship classId
+   * @param ecClassId class id of the custom change
+   * @param changeType insert, update or delete
+   * @param id ECInstanceID of the custom change
+   * @param sourceECInstanceId source ECInstanceId of the relationship
+   * @param targetECInstanceId target ECInstanceId of the relationship
+   */
+  public async addCustomRelationshipChange(
+    ecClassId: string,
+    changeType: SqliteChangeOp,
+    id: Id64String,
+    sourceECInstanceId: Id64String,
+    targetECInstanceId: Id64String
+  ): Promise<void> {
+    if (!this._ecClassIdsInitialized) await this.setupECClassIds();
+    if (this._relationshipSubclassIdsToSkip?.has(ecClassId)) return;
+    if (!this._relationshipSubclassIds?.has(ecClassId))
+      throw new Error(
+        `Misuse. id: ${id}, ecClassId: ${ecClassId} is not a relationship class. Use 'addCustomChange' instead.`
+      );
+
+    this._hasCustomChanges = true;
+    this._entityReferenceToCustomDataMap.set(
+      EntityReferences.fromEntityType(id, ConcreteEntityTypes.Relationship),
+      {
+        sourceIdOfRelationship: sourceECInstanceId,
+        targetIdOfRelationship: targetECInstanceId,
+        ecClassId,
+      }
+    );
+    this.handleChange(this.relationship, changeType, id);
+  }
+
+  public getClassFullNameFromECClassId(
```

Comment:
I think `getClassFullNameFromECClassId()` should not be implemented on  `ChangedInstanceIds` as public method. It could be some private method / variable, but it looks strange being public. Also, only populated classes can be resolved, not sure if it might introduce errors.

Reply:
Sample of class that would not be resolved: "BisCore::RepositoryModel"

Reply:
switched to private, added classFullName to the customdatamap for relationships